drivers/sensor/lsm303dlhc_accel/Kconfig

```diff
@@ -0,0 +1,134 @@
+# Kconfig - LSM303DLHC Three Axis Accelerometer configuration options
+
+#
+# Copyright (c) 2018 Philémon Jaermann
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig LSM303DLHC_ACCEL
+	bool
+	prompt "LSM303DLHC Three Axis Accelerometer"
+	depends on I2C
+	default n
+	help
+	  Enable driver for LSM303DLHC I2C-based triaxial accelerometer sensor.
+
+config LSM303DLHC_ACCEL_NAME
+	string
+	prompt "Driver name"
+	default "LSM303DLHC-accel"
+	depends on LSM303DLHC_ACCEL
+	help
+	  Device name with which the LSM303DLHC sensor is identified.
+
+config LSM303DLHC_ACCEL_I2C_MASTER_DEV_NAME
+	string
+	prompt "I2C master where LSM303DLHC is connected"
+	depends on LSM303DLHC_ACCEL
+	default "I2C_0"
+	help
+	  Specify the device name of the I2C master device to which LSM303DLHC is
+	  connected.
+
+config LSM303DLHC_ACCEL_I2C_ADDR
+	hex
+	prompt "LSM303DLHC accel I2C address"
+	depends on LSM303DLHC_ACCEL
+	default 0x19
+	help
+	  I2C address of LSM303DLHC acceleration sensor.	
+
+choice
+	prompt "Acceleration measurement range"
+	depends on LSM303DLHC_ACCEL
+	default LSM303DLHC_ACCEL_RANGE_2G
+	help
+	  Measurement full scale range for acceleration values.
+
+config LSM303DLHC_ACCEL_RANGE_2G
+	bool
+	prompt "+/-2g"
+
+config LSM303DLHC_ACCEL_RANGE_4G
+	bool
+	prompt "+/-4g"
+
+config LSM303DLHC_ACCEL_RANGE_8G
+	bool
+	prompt "+/-8g"
+
+config LSM303DLHC_ACCEL_RANGE_16G
+	bool
+	prompt "+/-16g"
+
+endchoice
+
+choice
+	prompt "Power mode"
+	depends on LSM303DLHC_ACCEL
+	default LSM303DLHC_POWER_MODE_NORMAL
+	help
+	  Choose between normal or low power operation mode for chip.
+
+config LSM303DLHC_POWER_MODE_NORMAL
+	bool
+	prompt "normal"
+
+config LSM303DLHC_POWER_MODE_LOW
+	bool
+	prompt "low"
+
+endchoice
+
+choice
+	prompt "Data rate frequency"
+	depends on LSM303DLHC_ACCEL
+	default LSM303DLHC_ACCEL_ODR_1
+	help
+	  Data rate frequency of acceleration data.
+
+config LSM303DLHC_ACCEL_ODR_1
```

Comment:
Would it be simpler to replace all these bool configs with one int config, and limit the range?

Reply:
It would be simpler yes. But by doing so, nothing prevents the user from entering an unsupported value. In this case, it will either result in a compilation error as ODR macro won't be defined or the driver using a default value. So should I use a default value (compilation always ok but may not necessarily correspond to what the user expected) or no default value (compilation only ok if correct value entered) ?

Reply:
I had similar comment as well. All the sensor drivers are done in the same way, but I do not like it personally.
Maybe you can define it as 'int' with a 'range' between 0 and 9, explaining what number is what freq.

Reply:
Yes this seems good to me! Sorry @MaureenHelm , by re reading this, I don't know why I didn t get what you were saying the first time :(